cmd/zq/zq.go

```diff
@@ -96,15 +97,57 @@ func New(f *flag.FlagSet) (charm.Command, error) {
 	return c, nil
 }
 
+func compileFilter(p ast.Proc) (filter.Filter, error) {
+	if fproc, ok := p.(*ast.FilterProc); ok {
+		return filter.Compile(fproc.Filter)
+	}
+	return nil, nil
+}
+
+func liftFilter(p ast.Proc) (ast.Proc, filter.Filter, error) {
```

Comment:
Nit: i think the AST transform and the filter compilation would be better as separate steps. first do the AST transform, returning a `(ast.Proc, ast.BooleanExpr, error)`, then compile the `ast.BooleanExpr` into a `filter.Filter`. Just a nit as this is just one isolated AST transform here.

Reply:
yeah, totally agree